src/LEGO.AsyncAPI/Models/AsyncApiComponents.cs

```diff
@@ -380,5 +383,367 @@ public void SerializeV2(IAsyncApiWriter writer)
 
             writer.WriteEndObject();
         }
+
+        public void SerializeV3(IAsyncApiWriter writer)
+        {
+            if (writer is null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+
+            // If references have been inlined we don't need the to render the components section
+            // however if they have cycles, then we will need a component rendered
+            if (writer.GetSettings().InlineLocalReferences)
+            {
+                var loops = writer.GetSettings().LoopDetector.Loops;
+                writer.WriteStartObject();
+                if (loops.TryGetValue(typeof(AsyncApiJsonSchemaReference), out List<object> schemas))
```

Comment:
should use the multiformat.schema rather than only jsonSchema - as avro can now be inlined here as well.